x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java

```diff
@@ -116,4 +119,20 @@ static DeprecationIssue indexNameCheck(IndexMetaData indexMetaData) {
         }
         return null;
     }
+
+    static DeprecationIssue nodeLeftDelayedTimeCheck(IndexMetaData indexMetaData) {
+        String setting = UnassignedInfo.INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING.getKey();
+        String value = indexMetaData.getSettings().get(setting);
+        if (Strings.isNullOrEmpty(value) == false) {
+            TimeValue parsedValue = TimeValue.parseTimeValue(value, setting);
+            if (parsedValue.getNanos() < 0) {
+                return new DeprecationIssue(DeprecationIssue.Level.WARNING,
+                    "Negative values for index.unassigned.node_left.delayed_timeout are deprecated and should be set to 0",
+                    "https://www.elastic.co/guide/en/elasticsearch/reference/master/breaking-changes-7.0.html" +
+                        "#_literal_index_unassigned_node_left_delayed_timeout_literal_may_no_longer_be_negative",
+                    "The index [" + indexMetaData.getIndex().getName() + "] is set to " + value);
```

Comment:
I think it's more clear if this details message reads something like:
`This index has [index.unassigned.node_left.delayed_timeout] set to [-43d], but negative values are not allowed`

Reply:
+1 to mentioning the setting name here too, but I think it's important to mention the name of the index as well.

Reply:
+1 added the setting name and left the index name.